apps/eth_geth_adapter/lib/eth_geth_adapter/balance.ex

```diff
@@ -0,0 +1,113 @@
+# Copyright 2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule EthGethAdapter.Balance do
+  @moduledoc false
+
+  alias Ethereumex.HttpClient, as: Client
+  alias EthGethAdapter.ERC20
+
+  import EthGethAdapter.Encoding
+
+  @doc """
+  Retrieve the balance of all given `contract_addresses` for the provided wallet `address`.
+  The `0x0000000000000000000000000000000000000000` address is used to represent Ether.
+  Any other given contract address will have their balance retrived on the corresponding
+  smart contract.
+
+  Returns a tuple of
+  ```
+  {
+    :ok,
+    %{
+      "contract_address_1" => integer_balance_1,
+      "contract_address_2" => integer_balance_2
+    }
+  }
+  ```
+  if successful or {:error, error_code} if failed.
+  """
+  def get(address, contract_address, block \\ "latest")
+
+  def get(address, contract_addresses, block) when is_list(contract_addresses) do
+    case ERC20.abi_balance_of(address) do
+      {:ok, encoded_abi_data} ->
+        contract_addresses
+        |> Enum.reduce([], fn contract_address, acc ->
```

Comment:
[`Enum.map/2`](https://hexdocs.pm/elixir/Enum.html#map/2) might give a little cleaner code also without need to reverse?